deps/chakrashim/src/jsrtutils.cc

```diff
@@ -1061,5 +1061,47 @@ JsValueRef CALLBACK CollectGarbage(
   return jsrt::GetUndefined();
 }
 
+void IdleGC(uv_prepare_t *prepareHandle) {
+  unsigned int nextIdleTicks;
+  CHAKRA_VERIFY(JsIdle(&nextIdleTicks) == JsNoError);
```

Comment:
spoke offline with @Yongqu . This will avoid calling CollectNow in EnterScriptEnd path.